src/cli.py

```
@@ -0,0 +1,71 @@
+import sys
+import click
+from pathlib import Path
+from src.comparsion import compare_and_output
+
+def validate_paths(input_path: str, output_path: str, whichtwo) -> None:
+    input_path = Path(input_path)
+    codeQL_directory = Path(input_path, "codeQL_data")
+    clojure_directory = Path(input_path, "clojure_data/atomsClojure.csv")
+    coccinelle_directory = Path(input_path, "coccinelle_data")
+
+    codeQL_exists = codeQL_directory.is_dir()
+    clojure_exists = clojure_directory.exists()
+    coccinelle_exists = coccinelle_directory.is_dir()
+
+    if not codeQL_exists and (whichtwo == "CQLvCNL" or whichtwo == "CLJvCQL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mCodeQL source does not exist: \033[0m{codeQL_directory}")
+    
+    if not clojure_exists and (whichtwo == "CLJvCNL" or whichtwo == "CLJvCQL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mClojure source does not exist: \033[0m{clojure_directory}")
+    
+    if not coccinelle_exists and (whichtwo == "CQLvCNL" or whichtwo == "CLJvCNL" or whichtwo == "all"):
+        raise FileNotFoundError(f"\033[91mCoccinelle source does not exist: \033[0m{coccinelle_directory}")
+    
+    if not whichtwo:
+        if codeQL_exists and clojure_exists and coccinelle_exists:
+            whichtwo = "all"
+        elif codeQL_exists and clojure_exists:
+            whichtwo = "CLJvCQL"
+        elif coccinelle_exists and clojure_exists:
+            whichtwo = "CLJvCNL"
+        elif codeQL_exists and coccinelle_exists:
+            whichtwo = "CQLvCNL"
+        else:
+            print(f"\033[91mWARNING: \033[0mOnly one or fewer input source for comparsion, and the output could be meaningless.")
```

We should handle errors consistently. In some cases, a `FileNotFoundError` is raise, while here, we print and exist early. There is a `try-except` in the command and that is good. Maybe raise an error here too? You can always define a custom error if raising `FileNotFoundError` does not make sense